FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc

```diff
@@ -212,9 +212,62 @@ std::unique_ptr<fastsim::Particle> fastsim::ParticleManager::nextGenParticle() {
       continue;
     }
 
-    // particle must be produced within the beampipe
-    if (productionVertex->position().perp2() * lengthUnitConversionFactor2_ > beamPipeRadius2_) {
-      continue;
+    // previously: particle must be produced within the beampipe
+    // now: particles which do not descend from exotics must be produced within the beampipe
+    int exoticRelativeId = 0;
+    if (productionVertex->position().perp2() * lengthUnitConversionFactor2_ > beamPipeRadius2_)  //
+    {
+      std::vector<HepMC::GenParticle*>::const_iterator relativesIterator_ =
+          productionVertex->particles_in_const_begin();
+      std::vector<HepMC::GenParticle*>::const_iterator relativesIteratorEnd_ =
+          productionVertex->particles_in_const_end();
+      bool hasExoticAssociation = false;
+      for (; relativesIterator_ != relativesIteratorEnd_; ++relativesIterator_) {
+        const HepMC::GenParticle& genRelative = **relativesIterator_;
```

Comment:
I think the modern loop is more readable (and shorter)
```
      bool hasExoticAssociation = false;
      for (const auto &relativesIterator_ : productionVertex->particles_in(particle)) {
        const HepMC::GenParticle& genRelative = *relativesIterator_;
```

If you use this version, you need to add `#include "HepMC/GenRanges.h"` in `ParticleManager.h`